### PR TITLE
Switch to TypeErrors 

### PR DESCRIPTION
**Pull request by ClaireHzl**
Switch from ValueError to TypeError when needed, and correction of some docstrings

### CODE REVIEW

**Review by glannuzel — CHANGES_REQUESTED** [review]

**Comment by ClaireHzl** [comment]
I added a function _check_goto_parameters for the GoToBased_Parts, to do the checks (because the function was too complex otherwise)

File: src/reachy2_sdk/parts/head.py [hunk 1]
```diff
@@ -222,6 +220,30 @@ def goto(
             self._wait_goto(response)
         return response
 
+    def _check_goto_parameters(self, duration: float, target: Any, q0: Optional[List[float]] = None) -> None:
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/head.py:223]: same :)

**Review by FabienDanieau — CHANGES_REQUESTED** [review]
Don't forget to add unit tests!

**Review by FabienDanieau — APPROVED** [review]
Thanks that's perfect!

**Review by glannuzel — APPROVED** [review]

**Pull request merged**